src/components/ContentSidebar/ContentSidebar.js

```diff
@@ -182,16 +186,40 @@ class ContentSidebar extends PureComponent<Props, State> {
      */
     componentWillReceiveProps(nextProps: Props): void {
         const { fileId, token }: Props = this.props;
-        const { fileId: newFileId, token: newToken, hasVersions }: Props = nextProps;
+        const { fileId: newFileId, token: newToken }: Props = nextProps;
 
         const hasTokenChanged = newToken !== token;
         const hasFileIdChanged = newFileId !== fileId;
         const currentFileId = newFileId || fileId;
 
         if (currentFileId && (hasTokenChanged || hasFileIdChanged)) {
-            this.fetchFile(currentFileId);
-            if (hasVersions) {
-                this.fetchVersions(currentFileId);
+            this.fetchData(nextProps, hasFileIdChanged);
+        }
+    }
+
+    /**
+     * Fetches the data for the sidebar
+     *
+     * @param {Object} Props the component props
+     * @param {boolean} hasFileIdChanged true if the file id has changed
+     */
+    fetchData({ fileId, hasVersions, hasActivityFeed, hasAccessStats }: Props, hasFileIdChanged?: boolean) {
+        if (hasFileIdChanged) {
+            this.setState({});
+        }
+
+        if (fileId) {
+            this.fetchFile(fileId);
+            if (hasAccessStats) {
+                this.fetchFileAccessStats(fileId);
+            }
+            if (hasActivityFeed) {
+                this.fetchComments(fileId, false);
+                this.fetchTasks(fileId);
+                this.fetchVersions(fileId, false);
+            } else if (hasVersions) {
```

Comment:
I am guessing `else if` because it only shows if there is no activity feed?

Reply:
Actually just caught a mistake there that my code doesnt match my comment. 

My thought was that we only need to fetch all the versions if there is an activity feed, else we can just do 1 call (potentially with a limit 1 or something), to render the version component in the details tab. I am now looking to see if this can be solved by adding a new field to the get file info call, then this else if can be avoided entirely.

Reply:
Looks like I can just use the ```version_number``` field to render sidebarVersions, so adding that field and removing the else if